conf/scalardb-cluster-custom-values-indirect-mode.yaml

```diff
@@ -19,14 +19,6 @@ envoy:
       service.beta.kubernetes.io/aws-load-balancer-internal: "true"
       service.beta.kubernetes.io/aws-load-balancer-type: "nlb"
 
-  resources:
```

Comment:
Just a question.
I'm wondering how much resource Envoy uses If they are not specified.
For memory, I think it will probably use just what it uses. CPU as well?

Reply:
@feeblefakie 
Thank you for your review and question!

> Just a question.
> I'm wondering how much resource Envoy uses If they are not specified.
> For memory, I think it will probably use just what it uses. CPU as well?

Yes. You are right.

If there is no `resources.limits` configuration, the pod will use resources (both CPU and Memory) as possible (i.e., it may use the resources until the worker node's upper limit).

When the Envoy pod and ScalarDB/ScalarDL pod are one-to-one, I think Envoy does not use resources more than ScalarDB/ScalarDL. In other words, I think ScalarDB/ScalarDL will become the bottleneck before Envoy.

But, I think your concern is correct. It's a very rare case, but there is a possibility that the Envoy consumes all resources on the worker node.

---

I think we can add `resources.limits` to the sample configuration that limits CPU and Memory to 2vCPU / 4GB.

There is no basis for these numbers (i.e., we cannot use these numbers for resource estimation), but we can avoid an unexpected issue that Envoy consumes all worker node resources.

What do you think?
If no problem, I will add the above limitation in the sample configuration.

---

FYI: 

Now I'm thinking about it might be better to add these resource limitations as default values on the Helm Chart side.

The `resources.requests` might prevent some developers' local tests (if the local environment does not have enough resources, they cannot deploy pods), but the `resources.limits` does not prevent local tests (the `limits` only limit resources, it does not require enough resources).

So, I think we can add the default value of `resources.limits` on the Helm Chart side. I will consider it as another task.